src/coinjoin/interfaces.cpp

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) 2024 The Dash Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <interfaces/coinjoin.h>
+
+#include <coinjoin/client.h>
+#include <wallet/wallet.h>
+
+#include <memory>
+#include <string>
+
+namespace coinjoin {
+namespace {
+
+class CoinJoinClientImpl : public interfaces::CoinJoin::Client
+{
+    CCoinJoinClientManager& m_clientman;
+
+public:
+    explicit CoinJoinClientImpl(CCoinJoinClientManager& clientman)
+        : m_clientman(clientman) {}
+
+    void resetCachedBlocks() override
+    {
+        m_clientman.nCachedNumBlocks = std::numeric_limits<int>::max();
```

Comment:
should probably be an optional?

Reply:
agree, but it's out of scope of this PR because this code is not new, it's moved from src/wallet/interfaces.cpp